packages/components/upload/src/ajax.ts

```diff
@@ -70,7 +70,8 @@ export const ajaxUpload: UploadRequestHandler = (option) => {
   const formData = new FormData()
   if (option.data) {
     for (const [key, value] of Object.entries(option.data)) {
-      if (isArray(value) && value.length) formData.append(key, ...value)
+      if (isArray(value) && value.length)
+        value.forEach((item) => formData.append(key, item))
```

Comment:
Since it's a type issue, we shouldn't change how the runtime code works. Refer to [this](https://github.com/element-plus/element-plus/pull/18331/commits/7922a03104cc2c77fb73dc7aaaf2429609c19eb4#diff-054875df38de531805c46c402c7943693d4e049f56c732f0be36133b4c01065cR22)